src/lib.rs

```diff
@@ -588,21 +676,35 @@ impl<T> StatefulWidget for Menu<T> {
             };
             let has_children = !item.children.is_empty();
 
-            let group_x_pos = x_pos;
+            let group_x_pos = {
+                if state.orientation.is_right() {
+                    x_pos
+                } else {
+                    let name_len = u16::try_from(item.name.len()).expect("name should be short");
+                    x_pos - self.drop_down_width + name_len
```

Comment:
x_pos is 0u16 for the first item, this substraction will panic.

Reply:
OMG, thanks for noticing this.
Unfortunately it would panic anyway since in this case the drop-down is being rendered to the left and so since there's no place for it anyway, the Clear widget will panic with this:
`Trying to access position outside the buffer`
I'll fix this one in a sec